app/src/main/kotlin/github/com/rhacco/dota2androidapp/viewmodel/MatchesViewModel.kt

```diff
@@ -0,0 +1,80 @@
+
+import android.app.Application
+import android.arch.lifecycle.AndroidViewModel
+import android.arch.lifecycle.MediatorLiveData
+import android.util.Log
+import github.com.rhacco.dota2androidapp.App
+import github.com.rhacco.dota2androidapp.R
+import github.com.rhacco.dota2androidapp.api.TopLiveGamesResponse
+import github.com.rhacco.dota2androidapp.lists.LiveMatchesItemData
+import github.com.rhacco.dota2androidapp.sources.repos.matches.RealtimeStatsRepository
+import github.com.rhacco.dota2androidapp.sources.repos.matches.TopLiveGamesRepository
+import io.reactivex.disposables.CompositeDisposable
+
+class MatchesViewModel(application: Application) : AndroidViewModel(application) {
+    private val mIsLoading = MediatorLiveData<Boolean>()
+    private val mDisposables = CompositeDisposable()
+    val mLiveMatchesQuery = MediatorLiveData<List<TopLiveGamesResponse.Game>>()
+    val mLiveMatchesItemDataQuery = MediatorLiveData<LiveMatchesItemData>()
+
+    override fun onCleared() = mDisposables.clear()
+
+    fun getLiveMatches() {
+        mIsLoading.value = true
+        mDisposables.add(TopLiveGamesRepository.getTopLiveGames()
+                .subscribe(
+                        { result ->
+                            mIsLoading.value = false
+                            mLiveMatchesQuery.value = result
+                        },
+                        { error ->
+                            mIsLoading.value = false
+                            Log.d(App.instance.getString(R.string.log_msg_debug),
+                                    "Failed to fetch top live games: " + error)
+                        }
+                ))
+    }
+
+    fun getLiveMatchesItemData(averageMMR: Int, serverSteamId: Long) {
+        mIsLoading.value = true
+        mDisposables.add(RealtimeStatsRepository.getRealtimeStats(serverSteamId)
+                .subscribe(
+                        { result ->
+                            mIsLoading.value = false
+                            val newItemData = LiveMatchesItemData()
+                            newItemData.mAverageMMR = averageMMR
+                            if (averageMMR < 1)
+                                newItemData.mTitle = "Tournament Match (Match ID " +
+                                        result.match.matchid + ")"
+                            else
+                                newItemData.mTitle = "Ranked Match (Average MMR " + averageMMR +
+                                        ", Match ID " + result.match.matchid + ")"
+                            if (result.teams.size == 2 &&
+                                    result.teams[0].players.size == 5 &&
+                                    result.teams[1].players.size == 5) {
+                                val radiantPlayers = result.teams[0].players
+                                val direPlayers = result.teams[1].players
+                                newItemData.mBlue = radiantPlayers[0].name
```

Comment:
I am not sure about calling players on radiant and dire by their color names. Color names I think should be considered a visual. Maybe say mRadiantPlayer0, mRadiantPlayer1 and comment the color names in the xml.

I can see why you did it: its within the context of anyone playing dota. But most people who play dota don't remember which color corresponds to raidant or dire and for that reason I don't think it shoudl be used.

Reply:
I agree, thanks. Done